src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,386 @@
+    return (
+      <>
+        <IcseFormGroup>
+          {/* vpn server name */}
+          <IcseNameInput
+            id={composedId}
+            className={classNameModalCheck}
+            componentName="vpn-server-name"
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() =>
+              this.props.invalidCallback(this.state, this.props)
+            }
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+            hideHelperText
+          />
+          {/* certificate_crn */}
+          <IcseTextInput
+            componentName="vpn-server-certificate-crn"
+            className={classNameModalCheck}
+            tooltip={{
+              content:
+                "Must use Secrets Manager to generate certificate_crn of secret for this VPN server.",
+              align: "top-left",
+            }}
+            id="certificate_crn"
+            labelText="Certificate CRN"
+            value={this.state.certificate_crn}
+            onChange={this.handleInputChange}
+            field="certificate_crn"
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          {/* method toggle */}
+          <IcseSelect
+            formName="vpn-server-method"
+            className={classNameModalCheck}
+            name="method"
+            labelText="Method"
+            groups={["certificate", "username"]}
+            value={this.state.method}
+            handleInputChange={this.handleInputChange}
+          />
+          {/* client_ca_crn or identity_provider */}
+          {this.state.method === "certificate" ? (
+            <IcseTextInput
+              componentName="vpn-server-client-ca-crn"
+              className={classNameModalCheck}
+              id="client_ca_crn"
+              labelText="Client CA CRN"
+              value={this.state.client_ca_crn}
+              onChange={this.handleInputChange}
+              field="client_ca_crn"
+            />
+          ) : (
+            <IcseTextInput
+              componentName="vpn-server-identity-provider"
+              className={classNameModalCheck}
+              id="identity_provider"
+              labelText="Identity Provider"
+              value={this.state.identity_provider}
+              onChange={this.handleInputChange}
+              field="identity_provider"
+            />
+          )}
+          {/* client_ip_pool */}
+          <IcseTextInput
+            id={this.props.data.name + "-client-ip-pool"}
+            componentName="vpn-server-client-ip-pool"
+            className={classNameModalCheck}
+            name="client_ip_pool"
+            field="client_ip_pool"
+            value={this.state.client_ip_pool}
+            placeholder="x.x.x.x"
+            labelText="Client IP Pool CIDR"
+            invalidCallback={() =>
+              isIpv4CidrOrAddress(this.state.client_ip_pool) === false
+            }
```

Comment:
one of your problems with the storybook is here, you can just run this function and pass the result as `invalid` instead of using `invalidCallback`


all other text inputs that dont have validation should take `invalid={false}` as a prop - after changing all those things, the storybook worked for me

Reply:
@jvallexm I'm a bit confused about this terminology, am I creating a callback that checks the overlap or is the user checking this before entering a value here? Can you give me an example of what a user might be passing in?